__tests__/graph.test.ts

```diff
@@ -91,4 +91,127 @@ d6.addEdge(node3, node1)
+    getPriority(): number {
+        return this.priority
+    }
+    hasPriorityOver (node: PrioritySetElement<number>): boolean {
+        return  this.priority < node.getPriority()
+    }
+    updateIfDuplicateOf (node: PrioritySetElement<number> | undefined): boolean {
+        if(node) {
+            return this.priority == node.getPriority()
+        }
+        return false
+    }
+}
+let ps0 = new PrioritySet<number>()
+test('test priority set', () => {
+    ps0.push(new SimpleElement(3))
+    ps0.push(new SimpleElement(5))
+    ps0.push(new SimpleElement(5))
+    ps0.push(new SimpleElement(7))
+    ps0.push(new SimpleElement(1))
+    ps0.push(new SimpleElement(4))
+    ps0.push(new SimpleElement(1))       
+    expect(ps0.size()).toBe(5)
+    expect(ps0.peek()?.getPriority()).toBe(1)
+    expect(ps0.pop()?.getPriority()).toBe(1)
+    ps0.empty()
+    expect(ps0.size()).toBe(0)
+
+})
+let d7 = new DependencyGraph<number>()
+let d8 = new DependencyGraph<number>()
+let d9 = new DependencyGraph<number>()
+test('test dependency graph', () => {
+    expect(d7.getEdges(node1).size).toBe(0)
+    d7.merge(d5)
+    expect(d7.size()).toStrictEqual(d5.size())
+
+    d8.addNode(node1)
+    d9.addEdge(node1, node2)
+    d8.merge(d9)
+    expect(d8.size()).toStrictEqual(d9.size())
+    expect(d9.getBackEdges(node2).size).toBe(1)
+    d8.removeNode(node2)
+    expect(d8.size()).toStrictEqual([1,0])
+})
+
+
+let d10 = new DependencyGraph<number>()
+test('test add/remove Edges', () => {
+    d10.addEdge(node1, node2)                   // {(node1 -> node2)}
+    expect(d10.size()).toStrictEqual([2, 1])
+
+    d10.addBackEdges(node2, new Set<number>().add(node1).add(node3))  // {(node1 -> node2), (node3 -> node2)}
+    expect(d10.size()).toStrictEqual([3, 2])
+
+    d10.addEdges(node1, new Set<number>().add(node2).add(node3).add(node4))      // {(node1 -> node2), (node1 -> node3), (node1 -> node4), (node3 -> node2)}
+    expect(d10.size()).toStrictEqual([4, 4])
+
+    d10.addEdges(node5, new Set<number>().add(node1))   // {(node1 -> node2), (node1 -> node3), (node1 -> node4), (node3 -> node2), {node5 -> node1}}
+    expect(d10.size()).toStrictEqual([5, 5])    
+
+    d10.removeEdge(node1, node2)        // {(node1 -> node3), (node1 -> node4), (node3 -> node2), {node5 -> node1}}
+    expect(d10.size()).toStrictEqual([5, 4])
+
+})
+
+let d11 = new DependencyGraph<number>()
+let d12 = new DependencyGraph<Object>()
+test('test the DOT representation of the dependency graph', () => {
+    expect(d11.toString()).toBe('digraph G {'+'\n}')
+
+    d11.addNode(node1)          // { node1 }
+    expect(d11.toString()).toBe('digraph G {\n"1";\n}')
+
+    d11.addEdge(node1, node2)   // { (node1 -> node2) }
+    expect(d11.toString()).toBe('digraph G {\n"1"->"2";\n}')
+
+    let obj = {0:1}
+    d12.addNode(obj)
+    expect(d12.toString()).toBe('digraph G {\n"[object Object]";\n}')
+
+    // TODO: Some issues are that it cannot detect cycles and overlap
```

Comment:
This comment is unclear. What is this referring to?

Reply:
The issue is that when we create a cycle like a line 180 ```(node1 -> node2), (node2 -> node1)```, it cannot detect cycle error. Also, cannot detect overlap in line 183 ```(node1 -> node2), (node1 -> node2)```.

Reply:
I solve the problem of this test in https://github.com/lf-lang/reactor-ts/pull/108/commits/5738ef26d5446e2a6f2951db94ebf4e99e501039